src/workloads/query/LookupColocatedData.yml

```diff
@@ -0,0 +1,368 @@
+        PhaseConfig:
+          Repeat: 10
+          Database: *Database
+          Operations:
+          - OperationMetricsName: {^Parameter: {Name: "MetricsName", Default: "Lookup"}}
+            OperationName: RunCommand
+            OperationCommand:
+              aggregate: {^Parameter: {Name: "Coll", Default: "localColl"}}
+              pipeline:
+                [
+                  {$lookup: {
+                    from: {^Parameter: {Name: "Coll", Default: "localColl"}},
+                    let: {local_field: "localField"},
+                    pipeline: [
+                      {$match: 
+                        {
+                          $expr: {$eq: ['$localField','$$local_field']}
+                        }
+                      }
+                    ],
+                    as: "joined"
+                  }},
+                ]
+              cursor: {batchSize: {^Parameter: {Name: "BatchSize", Default: 101}}}
+
+Actors:
+- Name: CreateShardedCollections
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *NumPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database
+        # This collection will have data distributed across 6 chunks.
+        - OperationMetricsName: ShardLocalCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: test.ShardedLocal
+            key: {*LocalShardKey: hashed}
+            numInitialChunks: 6
+        # Here we use 1 chunk. This ensures that all the data for this coll is on one shard.
+        - OperationMetricsName: ShardLocalCollectionAllOnOneShard
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: test.ShardedLocalAllOnOneShard
+            key: {*LocalShardKey: hashed}
+            numInitialChunks: 1
+        # Disable the balancer so that it can't skew results while the $lookups are running.
+        - OperationMetricsName: DisableBalancer
+          OperationName: AdminCommand
+          OperationCommand:
+            balancerStop: 1
+
+# On the sharded collections, an index wil automatically be created to support the shard key. Here
+# we create an equivalent index on the unsharded collection.
+- Name: CreateIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *NumPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Operations:
+        - OperationName: RunCommand
+          OperationCommand:
+            createIndexes: *UnshardedLocal
+            indexes:
+            - key: {*LocalShardKey: 1}
+              name: localShardKey_1
+
+- ActorFromTemplate:
+    TemplateName: LoadDocuments
+    TemplateParameters:
+      OnlyActiveInPhase: 2
```

Comment:
Can you elaborate on what these numbers do and why we need them?

Reply:
When specifying an actor in Genny, you need to indicate what the actor should be doing in each phase. It may be performing some operations, which need to be specified, or it could be doing nothing (no-op). It's a common pattern that an actor is only active in exactly one phase, and is doing a no-op in every other phase. [Here](https://github.com/mongodb/genny/blob/939d9bdbdc8830a1a2af4b27237743ef07a4cb09/src/workloads/query/LookupSBEPushdown.yml#L362-L374) is an example of doing that with more verbose syntax. Here, I'm using `OnlyActiveInPhase` as a more concise syntax for the same thing. So for example, this actor is only performing actions in phase 2.

If you wanted to write workloads where reads/writes happen concurrently, you could specify the same phase number for multiple actors. In our case, each query should be executed in a phase where no other operations are running. So the `OnlyActiveInPhase` numbers should be increasing throughout the file. 

I suppose all the inserts could happen in the same phase. I had it this way for developing/debugging purposes.